ui/apps/platform/src/Containers/VulnMgmt/Entity/Policy/VulnMgmtPolicyOverview.js

```diff
@@ -216,26 +214,70 @@ const VulnMgmtPolicyOverview = ({ data, entityContext, setRefreshTrigger }) => {
     } else {
         const getDeploymentTableColumns = getCurriedDeploymentTableColumns(isFeatureFlagEnabled);
 
+        const fragmentToUse = isFeatureFlagEnabled('ROX_POSTGRES_DATASTORE')
+            ? DEPLOYMENT_LIST_FRAGMENT_UPDATED
+            : DEPLOYMENT_LIST_FRAGMENT;
+
+        const failingDeploymentQuery = gql`
+            query getDeploymentsFailingForPolicy(
+                $id: ID!
+                $policyQuery: String
+                $deploymentQuery: String
+                $pagination: Pagination
+            ) {
+                policy(id: $id) {
+                    id
+                    deploymentCount(query: $deploymentQuery)
+                    deployments(query: $deploymentQuery, pagination: $pagination) {
+                        ...deploymentFields
+                    }
+                }
+            }
+            ${fragmentToUse}
+        `;
+
+        const fullEntityContext = workflowState.getEntityContext();
+        const failingDeploymentQueryVariables = {
+            id,
+            deploymentQuery: queryService.objectToWhereClause({
+                'Policy Violated': true,
+                ...queryService.entityContextToQueryObject(fullEntityContext),
+            }),
+            ...vulMgmtPolicyQuery,
+            pagination: queryService.getPagination(sort, page, 2 /* TODO Remove this param */),
+        };
+
         policyFindingsContent = (
-            <div className="pdf-page pdf-stretch pdf-new flex shadow rounded relativebg-base-100 mb-4 mx-4">
-                <TableWidget
-                    header={`${failingDeployments.length} ${pluralize(
+            <Query query={failingDeploymentQuery} variables={failingDeploymentQueryVariables}>
```

Comment:
I'm using the old-style render props pattern for the GQL query here since the child component that is affected is being conditionally rendered. The alternatives seemed to be:

- Always pull the deployment data via a top level `useQuery` hook, even when we don't need it
- Extract the `<TableWidget>` here out into a new wrapped component specifically for this use case and use the hook there

This seemed like the lighter-touch approach.